### PR TITLE
chore: render download progress in the host process

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/server/registry/oopDownloadBrowserMain.ts [hunk 1]
```diff
@@ -15,37 +15,45 @@
...
+
+function browserDirectoryToMarkerFilePath(browserDirectory: string): string {
+  return path.join(browserDirectory, 'INSTALLATION_COMPLETE');
+}
+
+function downloadFile(options: DownloadParams): Promise<void> {
   log(`running download:`);
-  log(`-- from url: ${url}`);
-  log(`-- to location: ${destinationPath}`);
+  log(`-- from url: ${options.url}`);
```
Comment (dgozman) [inline packages/playwright-core/src/server/registry/oopDownloadBrowserMain.ts:47]: It looks like these three lines are already logged in the main process.

File: packages/playwright-core/src/server/registry/browserFetcher.ts [hunk 2]
```diff
@@ -90,6 +95,20 @@ function downloadBrowserWithProgressBarOutOfProcess(title: string, browserDirect
...
+
+  debugLogger.log('install', `running download:`);
+  debugLogger.log('install', `-- from url: ${url}`);
+  debugLogger.log('install', `-- to location: ${zipPath}`);
+  const downloadParams: DownloadParams = {
+    title,
+    browserDirectory,
+    url,
+    zipPath,
+    executablePath,
```
Comment (dgozman) [inline packages/playwright-core/src/server/registry/browserFetcher.ts:107]: IIRC, passing `undefined` through IPC turns it into `null`? Let's make sure nothing breaks in this case.

File: packages/playwright-core/src/server/registry/oopDownloadBrowserMain.ts [hunk 3]
```diff
@@ -86,85 +94,33 @@ function downloadFile(url: string, destinationPath: string, options: DownloadFil
...
-  const totalRows = 10;
-  const stepWidth = 8;
-  let lastRow = -1;
-  return (downloadedBytes: number, totalBytes: number) => {
-    const percentage = downloadedBytes / totalBytes;
-    const row = Math.floor(totalRows * percentage);
-    if (row > lastRow) {
-      lastRow = row;
-      const percentageString = String(percentage * 100 | 0).padStart(3);
+process.on('message', async message => {
```
Comment (dgozman) [inline packages/playwright-core/src/server/registry/oopDownloadBrowserMain.ts:112]: While we are here, let's self-destruct on `process.on('disconnect')`?

**Review by dgozman — APPROVED** [review]

File: packages/playwright-core/src/server/registry/browserFetcher.ts (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/registry/browserFetcher.ts:107]: I would think this happens only to the top-level value.

**Pull request merged**